### PR TITLE
Ci/add actions

### PR DESCRIPTION
**Pull request by nateglims**
*Description of changes:* Add github actions. Based mostly on Jobs version.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          make -C build/ coverage
+          lcov --rc lcov_branch_coverage=1 --remove build/coverage.info '*test*' --output-file build/coverage.info
+          lcov --rc lcov_branch_coverage=1 --remove build/coverage.info '*CMakeCCompilerId*' --output-file build/coverage.info
+          lcov --rc lcov_branch_coverage=1 --remove build/coverage.info '*mocks*' --output-file build/coverage.info
+          lcov --list build/coverage.info
+      - name: lcov-cop
+        uses: ChicagoFlutter/lcov-cop@v1.0.2
+        with:
+          path: "build/coverage.info"
+          min_coverage: 95
```
Comment (aggarw13) [inline .github/workflows/ci.yml:39]: Full coverage requirement?
```suggestion
          min_coverage: 100
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          exclude: "**/*test*"
```
Comment (aggarw13) [inline .github/workflows/ci.yml:40]: Is this required when test information has been removed from the `coverage.info` file?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nateglims) [reply .github/workflows/ci.yml:39]: Won't pass

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nateglims) [reply .github/workflows/ci.yml:40]: Don't know, just copied from Jobs.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nateglims) [reply .github/workflows/ci.yml:39]: https://github.com/nateglims/device-defender-for-aws-iot-embedded-sdk/runs/1254916257?check_suite_focus=true#step:6:151

Comment (nateglims) [reply .github/workflows/ci.yml:39]: Is this remedied?

**Comment by nateglims** [comment]
An example of the run: https://github.com/nateglims/device-defender-for-aws-iot-embedded-sdk/actions/runs/307003664

**Pull request closed without merging**